### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by r-oung**
@TheW00Xer some comments:

I think it's important for you to develop some good coding habits. In general,

- Always include a `README.md` file at the root of the repository. It should include some basic information, e.g. what is this project? Later on, you will also want to include: how to setup the project, how to use the project, etc.
- Include comments for classes, functions, and non-trivial code

Always write code as if some other developer will be looking at it.

I'll talk to you about "Pull requests" next week. For now, don't merge this branch.

### CODE REVIEW

**Comment by r-oung** [comment]
In your project name `Temi_SimpleNavigation_App` you're mixing pascal-case with snake-case [ref](https://betterprogramming.pub/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841). Stick to one convention. Don't mix them.

**Comment by r-oung** [comment]
In `MainActivity.java`, the variable `TAG` is uninitialized.

Also, instead of `String TAG`, you can declare it at `private` because it's only going to be used in this class and you can declare it as `final` because the value will never change. So you can write `private final String TAG`. As your programs become more and more complex, it will become more important to scope variables.

**Comment by r-oung** [comment]
Just because the code "works" doesn't necessarily mean it's good code. 

In `MainActivity.java`, line 29 is never used. It's important to remove code that isn't used because this can confuse the reader of code reviews. In the bottom left-hand corner of Android Studio, there's a "Problems" tab. Try to understand and correct all problems.

**Comment by r-oung** [comment]
Avoid hard-coding the text in `activity_main.xml`, e.g. `TextView`, `Position A`, `Position B`, etc. Make use of the `strings.xml` file.

**Comment by r-oung** [comment]
@TheW00Xer When you have fixed all of the above issues, let me know here.

**Comment by TheW00Xer** [comment]
Thank you for the comments.
I've followed them and made some changes.
1. Project name changed to "TemiSimpleNavigationApp" - I was not aware of restrictions/rules when naming projects
2. Included README file and comments in the code
3. Variable TAG initialized like this:
```
public class MainActivity extends AppCompatActivity implements
        OnRobotReadyListener,
        OnCurrentPositionChangedListener {

    Robot mRobot;
    private final String TAG = "";
```
4. Text for buttons and TextView defined in Strings.xml
5. Code was also adjusted so there shouldn't be any line that is unused

I've tried to commit & push the changes but it asked me to merge it so I've cancelled the commit & push. For now the changes are saved just locally.

**Comment by r-oung** [comment]
>     private final String TAG = "";

You should initialize `TAG` with something, e.g. "MainActivity", "Debug", etc. Don't leave it empty like this.

> I've tried to commit & push the changes but it asked me to merge it so I've cancelled the commit & push. For now the changes are saved just locally.
It okay to merge. Make sure you commit and push all your changes to the `dev` branch. So if you're currently on the `master` branch, do the following to switch to the `dev` branch:
```shell
git pull
git checkout dev
```

**Comment by TheW00Xer** [comment]
@r-oung 
Updated file with fixed previous issues as suggested is was pushed to branch "dev" .
Please review it and let me know if you find any other issues or if you have other requests or tips for me.

File: app/src/main/java/com/example/simplenavigation/MainActivity.java [hunk 1]
```diff
@@ -17,30 +17,31 @@
 public class MainActivity extends AppCompatActivity implements
         OnRobotReadyListener,
         OnCurrentPositionChangedListener {
+
     Robot mRobot;
```
Comment (r-oung) [inline app/src/main/java/com/example/simplenavigation/MainActivity.java]: What about this variable? Should it be public? private? will it change in the future?

File: app/src/main/java/com/example/simplenavigation/MainActivity.java [hunk 2]
```diff
@@ -17,30 +17,31 @@
...
+    private final String TAG = MainActivity.class.getSimpleName();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
-        mRobot = Robot.getInstance();
-        TextView textView = findViewById(R.id.textViewPosition);
+        mRobot = Robot.getInstance();   //Here we initialize Robot instance
```
Comment (r-oung) [inline app/src/main/java/com/example/simplenavigation/MainActivity.java]: Remember, you don't need all that space between `;` and `//Here we ...`. One space is sufficient. i.e.
```Java
mRobot = Robot.getInstance(); //Here we initialize Robot instance
```

File: app/src/main/java/com/example/simplenavigation/MainActivity.java [hunk 3]
```diff
@@ -65,19 +66,22 @@ public void onClick(View view) {
     @Override
     protected void onStart() {
```
Comment (r-oung) [inline app/src/main/java/com/example/simplenavigation/MainActivity.java:87]: Please read about [Android Lifecycle](https://developer.android.com/guide/components/activities/activity-lifecycle) and understand what `onStart()`, `onStop()`, and `onCreate()` function represents.

File: app/src/main/java/com/example/simplenavigation/MainActivity.java [hunk 4]
```diff
@@ -65,19 +66,22 @@ public void onClick(View view) {
...
     protected void onStop() {
         super.onStop();
-
+        //Here we remove previously added event listeners in onStop function
         mRobot.removeOnRobotReadyListener(this);
         mRobot.removeOnCurrentPositionChangedListener(this);
     }
 
+    /**
+     * Here we add function "onRobotReady" that will hide our app's top bar when the parameter @param "isReady" is met
```
Comment (r-oung) [inline app/src/main/java/com/example/simplenavigation/MainActivity.java]: Keep your comments brief and concise. No need to say that `onRobotReady` is added -- it's obvious.

It's sufficient to write:
```
Hide the app's top bar when robot is ready
```

File: app/src/main/java/com/example/simplenavigation/MainActivity.java [hunk 5]
```diff
@@ -86,6 +90,10 @@ public void onRobotReady(boolean isReady) {
         }
     }
 
+    /**
+     * This function "onCurrentPositionChanged" uses parameter "position"
```
Comment (r-oung) [inline app/src/main/java/com/example/simplenavigation/MainActivity.java]: Yes, this function uses the `position` parameter. No need to write this. But what exactly is it doing?

**Comment by r-oung** [comment]
@TheW00Xer After you address the comments above, here's the next task:

Create 4 TextEdit boxes, i.e. X-position, Y-position, Yaw angle, Tilt angle (absolute value) and a button that commands the robot to go to that position and orient itself based on the yaw and tilt angles. When it arrives at the position, the robot should say, "I'm here"

**Comment by r-oung** [comment]
If you look below, you'll see that it says `This branch has conflicts that must be resolved`. Try to resolve the conflict on your own. If you don't understand, I'll show you later.

**Comment by TheW00Xer** [comment]
> @TheW00Xer After you address the comments above, here's the next task:
> 
> Create 4 TextEdit boxes, i.e. X-position, Y-position, Yaw angle, Tilt angle (absolute value) and a button that commands the robot to go to that position and orient itself based on the yaw and tilt angles. When it arrives at the position, the robot should say, "I'm here"

@r-oung 4 EditText boxes created and functional with "Go To Set coordinates" button.
Function TextToSpeech when robot arrives isn't working properly yet. Repository is updated.

**Review by r-oung — COMMENTED** [review]
My feedback

File: app/src/main/java/com/example/simplenavigation/MainActivity.java (hunk 5 above)
Comment (TheW00Xer) [reply app/src/main/java/com/example/simplenavigation/MainActivity.java]: Comment changed to
"Value of @param position is used to show User robot's current X,Y position and rotation & display tilt angle in TextView

File: app/src/main/java/com/example/simplenavigation/MainActivity.java (hunk 4 above)
Comment (TheW00Xer) [reply app/src/main/java/com/example/simplenavigation/MainActivity.java]: Comment changed to
"Hide app's top bar when the parameter @param "isReady" is met"

File: app/src/main/java/com/example/simplenavigation/MainActivity.java (hunk 2 above)
Comment (TheW00Xer) [reply app/src/main/java/com/example/simplenavigation/MainActivity.java]: Noted, changed spacing for all comments.

File: app/src/main/java/com/example/simplenavigation/MainActivity.java (hunk 1 above)
Comment (TheW00Xer) [reply app/src/main/java/com/example/simplenavigation/MainActivity.java]: Variable made Private.

**Pull request merged**